### PR TITLE
[BD-46] fix: remove ResizeObserver from useIndexOfLastVisibleChild hook

### PR DESCRIPTION
**Pull request by viktorrusakov**
## Description

Initially this PR aimed to fix `ResizeObserver` error that got reported in #1811, but turned out to be a good clean up opportunity for `useIndexOfLastVisibleChild` hook which resulted in removing `ResizeObserver` altogether, fixing the reported issue.

### Deploy Preview

https://deploy-preview-1842--paragon-openedx.netlify.app/components/hooks/useindexoflastvisiblechild/

**Note**: `ResizeObserver loop limit exceeded` error is not logged to browser's console by default (it still gets caught by New Relic though). To actually check that this error is not reproduced anymore do the following:

1. Go to the deploy-preview link above
2. Open your browser's console
3. Run `addEventListener('error', (e) => console.error(e))`
4. Try to resize your window in multiple ways

Notice that nothing gets logged to your console. If you do the steps above on [current version of useIndexOfLastVisibleChild hook](https://paragon-openedx.netlify.app/components/hooks/useindexoflastvisiblechild/) you will see multiple errors.

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @viktorrusakov!


When this pull request is ready, tag your edX technical lead.

File: src/Bubble/index.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const Bubble = React.forwardRef<HTMLDivElement, BubbleProps>(({
 
 Bubble.propTypes = {
   /** Specifies contents of the component. */
-  children: PropTypes.node.isRequired,
+  children: PropTypes.node,
```
Comment (viktorrusakov) [inline src/Bubble/index.tsx:41]: we actually have such a case in `Tabs` component, where `Bubble` is rendered without children, caught the error in dev mode

File: src/hooks/useIndexOfLastVisibleChild.mdx [hunk 2]
```diff
@@ -35,30 +35,40 @@ of the children until they exceed the width of the container.
   const children = useMemo(() => {
     const indexOfOverflowStart = indexOfLastVisibleChild + 1;
     const childrenList = elements.map((element, i) => (
-      <div className="px-4 pt-2" style={i >= indexOfOverflowStart ? invisibleStyles : {}}>{element}</div>
+      <div
```
Comment (viktorrusakov) [inline src/hooks/useIndexOfLastVisibleChild.mdx:38]: all the changes in this file are just a bit of clean up, had a lot of warnings in console about `key` prop and other props stuff (like missing props etc.)

**Review by adamstankiewicz — APPROVED** [review]
LGTM! Behavior seems to have stayed the same as far as I can tell as well.

**Comment by openedx-webhooks** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1842&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2022-12-12+14:58&entry.1917517419=2022-12-16+13:52&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**